### PR TITLE
[ENG-6650][ENG-6729] Remote Computing / Boa addon

### PR DESCRIPTION
**Pull request by felliott**
* Add infrastructure for new type of addons, `remote-computing`.
* Add implementation of first remote-computing service, Boa.
* ~GLEEP! I forgot to regenerate the migration. The migration is correct, but the order is wrong!~
* ~ERRRPP! Now there is a broken test.~

### CODE REVIEW

File: addon_service/credentials/models.py [hunk 1]
```diff
@@ -104,6 +105,9 @@ def authorized_accounts(self):
                 )
             ]
         except ExternalCredentials.authorized_storage_account.RelatedObjectDoesNotExist:
+            # FIXME: I have doubts that auth_storage_acct is correct. Shouldn't it be storage?
+            # FIXME: Where is citation and remote_computing?
+            # FIXME: does that docstring need to be updated?
             return None
```
Comment (opaduchak) [inline addon_service/credentials/models.py:111]: I suppose that this except clause is redundant

Comment (opaduchak) [reply addon_service/credentials/models.py:111]: But on the other hand, authorized account property accounts for all addon types

**Review by opaduchak — COMMENTED** [review]
Great work!
I've left some questions, but I am not sure whether any changes are required

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One potential change from me.

File: addon_service/addon_operation_invocation/views.py [hunk 2]
```diff
@@ -51,19 +55,27 @@ def retrieve_related(self, request, *args, **kwargs):
                 serializer = AuthorizedStorageAccountSerializer(
                     instance, context={"request": request}
                 )
-            else:
+            elif hasattr(instance, "authorizedcitationaccount"):
                 serializer = AuthorizedCitationAccountSerializer(
                     instance, context={"request": request}
                 )
+            else:  # computing
```
Comment (brianjgeiger) [inline addon_service/addon_operation_invocation/views.py:62]: Same comment here.

File: addon_service/configured_addon/computing/serializers.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+from addon_service.external_service.computing.models import ExternalComputingService
+
+
+RESOURCE_TYPE = get_resource_type_from_model(ConfiguredComputingAddon)
+
+
+class ConfiguredComputingAddonSerializer(ConfiguredAddonSerializer):
+    """api serializer for the `ConfiguredComputingAddon` model"""
+
+    root_folder = serializers.CharField(required=False, allow_blank=True)
```
Comment (opaduchak) [inline addon_service/configured_addon/computing/serializers.py:20]: I beleive this field is redundant for computing addons

**Review by opaduchak — CHANGES_REQUESTED** [review]
Seems good,
But I believe you have missed removing one extra field from configured compute addon serializer

File: addon_service/configured_addon/computing/views.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from addon_service.configured_addon.views import ConfiguredAddonViewSet
+
+from .models import ConfiguredComputingAddon
+from .serializers import ConfiguredComputingAddonSerializer
+
+
+class ConfiguredComputingAddonViewSet(ConfiguredAddonViewSet):
+    queryset = ConfiguredComputingAddon.objects.active()
+    serializer_class = ConfiguredComputingAddonSerializer
```
Comment (opaduchak) [inline addon_service/configured_addon/computing/views.py:9]: If you are planning to fetch credentials from `gravyvalet` for computing addons (in this PR), you should add action for credential fetching similar to `ConfiguredStorageAddonViewSet`

**Review by brianjgeiger — APPROVED** [review]
I'm good once Oleh is happy with this. Also, based on [this conversation](https://github.com/CenterForOpenScience/osf.io/pull/10851/files#r1883955318), are we setting the `wb_key` properly in this PR?

**Review by opaduchak — APPROVED** [review]
I have nothing to add, LGTM!

**Pull request merged**